File: src/Entities/Medicine.Entities/Models/Course.cs
using Medicine.Entities.Enums;
using Medicine.Entities.Models.Base;
using Medicine.Entities.Models.Translated;

namespace Medicine.Entities.Models
{
    public class Course : TranslationEntity<TranslatedCourse>
    {
        public int TherapyId { get; set; }
        public Therapy Therapy { get; set; }
        public CourseGroup? CourseGroup { get; set; }
        public int? CourseGroupID { get; set; }
        public CourseType CourseType { get; set; } = CourseType.None;
        public List<CourseSettings> CourseSettings { get; set; } = new List<CourseSettings>();
        public List<DosingFrequency> DosingFrequencies { get; set; } = new List<DosingFrequency>();
    }
}

File: front/medicine/src/app/shared/models/entity/Course.ts
import { CourseType } from "../../enums/CourseType";
import { User } from "./User";
import { DosingFrequency } from "./DosingFrequency";
import { Therapy } from "./Therapy";
import { CourseGroup } from "./CourseGroup";
import { CourseSettings } from "./CourseSettings";
import { TranslatedCourse } from "./translate/TranslatedCourse";


export class Course {
  therapyId!: number;
  therapy!: Therapy;
  courseGroup!: CourseGroup;
  courseGroupID!: number;
  courseType!: CourseType;
  courseSettings: CourseSettings[] = [];
  dosingFrequencies: DosingFrequency[] = [];
  translations: TranslatedCourse[] = [];
  user?: User;
  userId?: number;
  id!: number;
  createdAt?: Date;
}
